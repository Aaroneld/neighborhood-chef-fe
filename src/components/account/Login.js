import React from "react";
import { Link } from 'react-router-dom';
import { buttonStyles } from "../../styles";
import ls from 'local-storage';
import { nanoid } from 'nanoid';

const qs = require('querystring');
const crypto = require('crypto-browserify');

const Login = () => {
  const classes = buttonStyles();

  const handleClick = e => {
    e.preventDefault();
    ls.set('code_verifier', nanoid(43) );

    const hash = crypto.createHash('sha256')
                        .update(ls.get("code_verifier"))
                        .digest('base64')
                        .replace(/\+/g, '-')
                        .replace(/\//g, '_')
                        .replace(/=/g, '');  
    const query = {
        client_id: "0oa9qxckmhGd2aLsZ4x6",
        response_type: "code",
        scope: "openid",
        redirect_uri: "http://localhost:3000/login-redirect-url",
        state: "state-bsaWCD8F0dkd85REyU87",
        code_challenge_method: "S256",
        code_challenge: hash

    }

    const redirectURL = `https://dev-599411.okta.com/oauth2/default/v1/authorize?${qs.stringify(query)}`;
    console.log(hash);
    window.location.replace(`${redirectURL}`);
}

  return (
    <div className="login-container">
      <h1>Neighborhood Chef</h1>
      <p style={{ fontStyle: "italic" }}>Prepare to eat well</p>

      {/*need to add proper okta url for login here */}
        {" "}
        <div className={`${classes.root} ${classes.single}`} onClick={handleClick}>
          {" "}
          Login Securely with Okta
        </div>
      <div style={{ marginTop: "10px" }}>
        Don't have an account?{" "}
        <span className="green-link">
          <Link to="/register">Register now.</Link>
        </span>
      </div>
    </div>
  );
};

export default Login;
