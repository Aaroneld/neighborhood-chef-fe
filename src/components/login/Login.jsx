import React from 'react';
import { useHistory } from 'react-router-dom';
import AuthHeader from '../shared/AuthHeader';
import bgImage from '../../assets/chef.svg';
import Card from '@material-ui/core/Card';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { landingPageStyles } from './login.styles';
import ls from 'local-storage';
import { nanoid } from 'nanoid';

const qs = require('querystring');
const crypto = require('crypto-browserify');

const Login = () => {
  const styles = landingPageStyles({ bgImage });
  const { push } = useHistory();

  const handleClick = (e) => {
    e.preventDefault();
    ls.set('code_verifier', nanoid(43));

    const hash = crypto
      .createHash('sha256')
      .update(ls.get('code_verifier'))
      .digest('base64')
      .replace(/\+/g, '-')
      .replace(/\//g, '_')
      .replace(/=/g, '');
    const query = {
      client_id: process.env.REACT_APP_CLIENT_ID,
      response_type: 'code',
      scope: 'openid',
      redirect_uri: `${process.env.REACT_APP_FRONT_END_BASE_URL}/login-redirect-url`,
      state: 'state-bsaWCD8F0dkd85REyU87',
      code_challenge_method: 'S256',
      code_challenge: hash,
    };

    const redirectURL = `https://dev-36383529.okta.com/oauth2/default/v1/authorize?${qs.stringify(query)}`;
    window.location.replace(`${redirectURL}`);
  };

  return (
    <div className={styles.container}>
      <AuthHeader />
      <div className={styles.landingPageContainer}>
        <Card className="card">
          <Typography variant="h4">Let's start dining with the <span id="community">community!</span></Typography>
          <Typography variant="h6">
            Welcome to Neighborhood Chef, a powerful new tool to help you engage with your community, make
            friends, plan culinary related events, and most importantly, eat good food.
          </Typography>
          <Typography variant="h6" id="second-paragraph">
            It is our mission to help people engage with others in a safe and inclusive manner. Community is an 
            important aspect of social life and food has long been the medium through which this is shared. It is our hope that this service 
            is a small contributions to help <b>YOU</b> build your own. 
          </Typography>
          <Typography id="bonAppetit" variant="h6">
            ...Bon app√©tit!
          </Typography>
          <div className="buttons">
            <Button id="register" onClick={() => push('/register')}>
              Register
            </Button>
            <Typography variant="h6">Already have an account?</Typography>
            <Button id="login" onClick={handleClick}>
              Login
            </Button>
          </div>
        </Card>
      </div>
    </div>
  );
};

export default Login;
