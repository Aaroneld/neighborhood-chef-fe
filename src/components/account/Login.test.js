import React from "react";
import Login from "./Login.js";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("Test login static properties", () => {
  let LoginComponent;
  beforeEach(() => {
    LoginComponent = render(<Login />);
  });

  test("Login component renders", () => {
    expect(LoginComponent.getByText(/Login Component/i));
  });
});
