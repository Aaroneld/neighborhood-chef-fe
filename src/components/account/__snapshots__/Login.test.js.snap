// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders 1`] = `
<h1>
  Login Component
</h1>
`;
